src/main/java/net/openhft/chronicle/queue/impl/single/TableStoreWriteLock.java

```diff
@@ -66,13 +73,18 @@ public void lock() {
         assert checkNotAlreadyLocked();
 
         long currentLockValue = 0;
+        long currentLockTidValue = 0;
         TimingPauser tlPauser = pauser.get();
         try {
             currentLockValue = lock.getVolatileValue();
+            currentLockTidValue = lockTid.getVolatileValue();
+
             while (!lock.compareAndSwapValue(UNLOCKED, PID)) {
                 currentLockValue = lockGetCurrentLockValue(tlPauser);
             }
-
+            // Set tid explicitly here
+            lockedByThread = Thread.currentThread();
```

Comment:
`Thread.currentThread()` can be slow - maybe we need to hide this behind a feature property. Maybe we can even piggy back on one of the existing ones.

Reply:
actually, as discussed, lets re-use existing lockedByThread and enable with jvm.resource.tracing